safe_init/secrets.py

```diff
@@ -45,31 +93,109 @@ def resolve_secrets() -> Mapping[str, str | None]:
             if not common_secret_arn_prefix or secret_arn.startswith(common_secret_arn_prefix)
             else f"{common_secret_arn_prefix}{secret_arn}"
         )
+    return secret_arns
+
+
+def get_secrets_from_cache(secret_arns: dict[str, str]) -> tuple[dict[str, str], list[str]]:
+    """
+    Retrieves secrets from the cache and identifies any that are missing.
+
+    Args:
+        secret_arns (Dict[str, str]): A dictionary mapping secret names to their ARNs.
+
+    Returns:
+        A tuple containing:
+        - A dictionary of secrets retrieved from the cache.
+        - A list of ARNs for secrets that are not in the cache.
+    """
+    secrets_in_cache = {}
+    secrets_not_in_cache = []
 
+    for _, secret_arn in secret_arns.items():
+        secret_value = get_secret_from_cache(_strip_json_key_prefix_if_present(secret_arn))
```

Comment:
Can't we also use batch get here? Redis has support for it.

Reply:
Also, can't we "secrets_client" inside "get_secret_from_cache"?

Reply:
> Also, can't we "secrets_client" inside "get_secret_from_cache"?

What's the benefit? We currently have a clear separation of functions responsible for fetching secrets — one from AWS, one from cache — and it allows the function implementing the logic to decide which one to use